cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1871,21 +1933,22 @@ public void setSubscriptionCountry(String country) {
     }
 
     public Set<Notification> getNotifications() {
-		Gson gson = new Gson();
+        Gson gson = new Gson();
         SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CleverPush.context);
 
-		String notificationsJson = sharedPreferences.getString(CleverPushPreferences.NOTIFICATIONS_JSON, null);
-		if (notificationsJson != null) {
-			try {
-				Type type = new TypeToken<List<Notification>>() {}.getType();
-				List<Notification> notifications = gson.fromJson(notificationsJson, type);
-				return new HashSet<>(notifications);
-			} catch (Exception ex) {
-				Log.e("CleverPush", "error while getting stored notifications", ex);
-			}
-		}
-
-		// deprecated
+        String notificationsJson = sharedPreferences.getString(CleverPushPreferences.NOTIFICATIONS_JSON, null);
+        if (notificationsJson != null) {
+            try {
+                Type type = new TypeToken<List<Notification>>() {
+                }.getType();
+                List<Notification> notifications = gson.fromJson(notificationsJson, type);
```

Comment:
this is not needed, you don't need to do this as a TypeToken. 
create a model extending List<Notification> instead, and then do gson.fromJson(notificationsJson, CreatedModel.class);

also why did you choose a hashset here?